app/models/user.rb

```diff
@@ -20,5 +22,24 @@ def digest string
              end
       BCrypt::Password.create string, cost: cost
     end
+
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_column :remember_digest, User.digest(remember_token)
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil?
```

Comment:
tìm hiểu khác nhau giữa unless và nil

Reply:
Dạ unless là if not. E sửa thành unless r ạ